### PR TITLE
docs: add pull request review guidelines

### PR DESCRIPTION
**Pull request by StaberindeZA**
Because:

* The Subscription Platform team discussed pull request review guidelines during their work week in December 2022.

This commit:

* Adds the pr review guidelines to ecosystem docs.

Closes #FXA-6763

### CODE REVIEW

**Comment by StaberindeZA** [comment]
For context, this is the original presentation this document was drawn from.

https://docs.google.com/presentation/d/1ndT0W6-76D6imMwBbOmAGx00HZVMFywsS-nuufFjXLM/edit?usp=sharing

**Comment by LZoog** [comment]
@StaberindeZA is this intended for SubPlat only, or FxA + SubPlat?

**Comment by StaberindeZA** [comment]
> @StaberindeZA is this intended for SubPlat only, or FxA + SubPlat?

Hi @LZoog, at the moment we've only discussed it within SubPlat, so there are probably some SubPlat specific points in there which might not be useful for FxA. However, I think it'd be great if we can work this towards applying to both teams. What do you think?

File: docs/reference/team-processes/pull-request-review-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+## Pull requests are hard
+
+The guidelines provided below will hopefully make your pull requests reviews a little easier.
+
+## Why we do reviews
+
+- Reduce churn and ship faster
+- Reduce regressions
+  - The better the pull request review, the less likely it is that a regression
```
[the remark this answers is not in the document]
Comment (StaberindeZA) [reply docs/reference/team-processes/pull-request-review-guidelines.md:13]: Sounds good. I've updated. Thank you

File: docs/reference/team-processes/pull-request-review-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    it be broken up into smaller, more reusable modules?
+- Test the changes locally
+- Be precise about your comments
+- Keep your Ego Out of Code Reviews
+- Direct communication
+  - Consider scheduling a code review call, or discussing the pull request via Slack DM
+  - Ensure to add important notes discussed on calls, back into the pull request
+- Do not provide style notes
+  - Style notes should be enforced by the linter or style guide. If you would like to
+    change a style, create a new pull request to update the linter or style guide
```
Comment (biancadanforth) [inline docs/reference/team-processes/pull-request-review-guidelines.md:65]: We could link to the newly created front-end and back-end style guides?
React: https://mozilla.github.io/ecosystem-platform/reference/style-guides/react-style-guide
Node: https://mozilla.github.io/ecosystem-platform/reference/style-guides/node-style-guide

**Review by biancadanforth — APPROVED** [review]
Hi Reino; this LGTM; I made some suggestions in-line, but I also have a few higher level asks:
1. Did you discuss this guide with anyone from FxA?
  a. Since only the SubPlat team participated in the initial brainstorming session, I think it'd be good to either get an FxA reviewer or mark this as the guide created by/for the SubPlat team.
2. Is there a ticket for this effort? If not can you file one and reference it here?
  b. I'm mostly asking for this, because otherwise this work by you and by me as the reviewer isn't tracked or documented in a centralized place.

File: docs/reference/team-processes/pull-request-review-guidelines.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+title: Pull Request review guidelines
+---
+
+## Pull requests are hard
+
+The guidelines provided below will hopefully make your pull requests reviews a little easier.
+
+## Why we do reviews
+
+- Reduce churn and ship faster
```
[the remark this answers is not in the document]
Comment (StaberindeZA) [reply docs/reference/team-processes/pull-request-review-guidelines.md:11]: Mostly I was going off this definition of churn, listed below.

```
What Code Churn Is. Code churn, or code that is rewritten or deleted shortly after being written...
```

Specifically I was thinking of situations where code is merged, and bugs only identified during QA or in Production, thus increasing the turnaround time to fix the issue, compared to identifying the bug during PR review.

[the remark this answers is not in the document]
Comment (StaberindeZA) [reply docs/reference/team-processes/pull-request-review-guidelines.md]: Maybe change it to "long running code churn"?

[the remark this answers is not in the document]
Comment (biancadanforth) [reply docs/reference/team-processes/pull-request-review-guidelines.md]: Ah okay thanks for the clarification. I'd heard of churn before, but not with respect to code. Makes sense, and sure I think people will get it if you precede it with "code", heh. Or even if you link to that nice definition you shared.

**Comment by biancadanforth** [comment]
> > Hi Reino; this LGTM; I made some suggestions in-line, but I also have a few higher level asks:
> > ```
> > 1. Did you discuss this guide with anyone from FxA?
> >    a. Since only the SubPlat team participated in the initial brainstorming session, I think it'd be good to either get an FxA reviewer or mark this as the guide created by/for the SubPlat team.
> > 
> > 2. Is there a ticket for this effort? If not can you file one and reference it here?
> >    b. I'm mostly asking for this, because otherwise this work by you and by me as the reviewer isn't tracked or documented in a centralized place.
> > ```
> 
> @biancadanforth , thank you so much for taking a look and for your feedback and questions. I replied inline to your comments. For the other 2, please see below.
> 
>     1. I did not discuss this with anyone from FxA, except for my reply to Lauren's question on this PR. For now I've added a "Note" banner to the top of the page to indicate that this page applies to SubPlat only.
> 
>     2. Ah thank you for this note. I've created FXA-6763 and added you as a collaborator.

Sounds good thanks.

**Pull request merged**